content/en/blog/_posts/2023-09-15-userns-stateful-pods/index.md

```diff
@@ -0,0 +1,113 @@
+longer `UserNamespacesStatelessPodsSupport` but from 1.28 onwards you should
+use `UserNamespacesSupport`. There were many changes done and the requirements
+on the node hosts changed. So we took the opportunity to rename the feature
+flag.
+
+[volume-limitations]: https://v1-25.docs.kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#limitations
+
+## Demo
+
+We've created a demo which exploits [CVE 2022-0492][cve-link] and we show how
+the exploit occurs without user namespaces and how it is not possible when user
+namespaces are in use.
+
+This CVE is rated **HIGH** and allows **a container with no special privileges to
+read/write to any path on the host** and launch processes as root on the host
+too.
+
+{{< youtube id="" title="User Namespaces mitigating CVE-2022-0492 on Kubernetes">}}
+
+Most applications in containers run as root today, or as the same user (UID
+65534 is a somewhat popular choice). When this feature is enabled, it switches
+them to run as unprivileged users, with no changes needed in the app.
+
+This means two containers running as user 65534 will effectively be mapped to
+different users on the host, limiting what they can do to each other in case of
+an escape, and if they are running as root, the privileges on the host are
+reduced to the one of an unprivileged user.
+
+[cve-link]: https://unit42.paloaltonetworks.com/cve-2022-0492-cgroups/
+
+## Stack requirements
+
+There are requirements on the Linux kernel version as well as the container
+runtime to use this feature.
+
+On Linux you need Linux 6.3 or greater. This is because we rely on a kernel
+feature named idmap mounts, and we merged support to use idmap mounts with
+tmpfs in Linux 6.3.
+
+If you are using CRI-O with crun, this is [supported in CRI-O
+1.28][CRIO-release]. If you are using CRI-O with runc, this is still not
+supported.
+
+containerd support is currently targeted for containerd 2.0, doesn’t matter if
+you use it with crun or runc.
+
+Please note that contianerd 1.7 added _experimental_ support for user
+namespaces as implemented in Kubernetes 1.25 and 1.26. The redesign we did in
+1.27 is not supported by containerd 1.7, therefore it only works, in terms of
+user namespaces support, with Kubernetes 1.25 and 1.26.
+
+One limitation present in containerd 1.7 is that it needs to chown the rootfs
+of the container at container startup when using user namespaces. This means it
+has a storage overhead and can significantly impact the container startup
+latency. For containerd 2.0 we are working on an implementation that will
+eliminate the startup latency added and the storage overhead. Take this into
+account if you plan to use containerd 1.7 with user namespaces in production.
+
+None of these containerd limitations apply to [CRI-O 1.28][CRIO-release].
+
+[CRIO-release]: https://github.com/cri-o/cri-o/releases/tag/v1.28.0
+
+## What’s next?
+
+For Kubernetes 1.29 we are working with SIG-auth to integrate user namespaces
```

Comment:
```suggestion
Looking ahead to Kubernetes 1.29 we are working with SIG Auth to integrate user namespaces
```



Does “we” mean Rodrigo Campos Catelin, Giuseppe Scrivano and Sascha Grunet? If not, reword this.

Reply:
Fixed, thanks!